src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -153,15 +169,25 @@ const PlrDaoStakingTransactionBlock = ({
   const [totalKeyBasedPLRTokens, setTotalKeyBasedPLRTokens] = useState<number>(0);
   const [totalSmartWalletPLRTokens, setTotalSmartWalletPLRTokens] = useState<number>(0);
 
+  const [selectedRoute, setSelectedRoute] = useState<SelectOption | null>(null);
+  const [availableRoutes, setAvailableRoutes] = useState<Route[]>([]);
+  const [isLoadingAvailableRoutes, setIsLoadingAvailableRoutes] = useState<boolean>(false);
+
   const [accounts, setAccounts] = useState<AccountBalance[]>([]);
   const [isNFTMember, setIsNFTMember] = useState<boolean>(false);
 
   const hasEnoughPLR =
     totalKeyBasedPLRTokens >= MAX_PLR_TOKEN_LIMIT || totalSmartWalletPLRTokens >= MAX_PLR_TOKEN_LIMIT;
-
-  const filteredSupportedChains = supportedChains.filter((chain) =>
-    [CHAIN_ID.ETHEREUM_MAINNET, CHAIN_ID.POLYGON, CHAIN_ID.XDAI, CHAIN_ID.BINANCE].includes(chain.chainId)
+  const isPolygonAccountWithEnoughPLR = accounts.some(
+    (account) =>
+      account.chainName === 'Polygon' &&
```

Comment:
I know this is valid in code, but just always remember that checking string that is a title or a name is very bad pattern that might lead to issues if chain name changes (probably most of top chains except Ethereum had their names changed in last years), this is ok unless there are no other options, but here we can have plenty of options such as adding chainId instead of name to account and then pulling chain title later when we need it in UI, but here we can check against chain ID then

hope it makes sense